lib/ConfigReader/ConfigFile.cpp

```diff
@@ -0,0 +1,176 @@
+#include "ConfigFile.h"
+
```

Comment:
General comment: Looks good, however, I am pretty sure there is a library that can deal with ini / json / yml files without re-inventing the wheel (or since you use it as a key-value store only, there might be something more performant available..)

Reply:
Give some hints. I found only large libraries or big classes

Reply:
Well, we are already using the `cjson` library: https://github.com/SolarEnergyRacers/solar_car_control_system/blob/8abb118b1d73dade72f8b8aeac81eb4a83466d8b/lib/CarState/CarState.cpp

And for the `ini` option, there are libraries available (https://registry.platformio.org/search?q=ini&t=library) however, I did not work with any of these.

Generally speaking, if you already invested quite some time and it works stable with the current implementation, I suggest to go ahead with it, and think about again as soon as serious issues arise.

Reply:
" I suggest to go ahead with it" me too: Only 150 lines.